File: connectors/connector-trello/src/main/scala/com/taskadapter/connector/trello/TrelloConfig.scala
package com.taskadapter.connector.trello

import java.util

import com.taskadapter.connector.Priorities
import com.taskadapter.connector.definition.ConnectorConfig

import scala.beans.BeanProperty

class TrelloConfig(@BeanProperty var boardId: String, @BeanProperty var boardName: String) extends ConnectorConfig {

  setPriorities(createDefaultPriorities)

  /**
    * Creates default priorities.
    *
    * @return default priorities.
    */
  def createDefaultPriorities: Priorities = {
    val result: util.Map[String, Integer] = new util.HashMap[String, Integer]
    result.put("Lowest", 100)
    result.put("Low", 300)
    result.put("Medium", 500)
    result.put("High", 700)
    result.put("Highest", 1000)
    new Priorities(result)
  }

}

File: connectors/connector-trello/src/main/java/com/taskadapter/connector/trello/TrelloConfig.java
package com.taskadapter.connector.trello;

import com.taskadapter.connector.definition.ConnectorConfig;

import java.util.HashMap;
import java.util.Map;

public class TrelloConfig extends ConnectorConfig {

    private static final Map<String, Integer> DEFAULT_PRIORITIES = new HashMap<>();

    static {
        DEFAULT_PRIORITIES.put("Low", 100);
        DEFAULT_PRIORITIES.put("Normal", 500);
        DEFAULT_PRIORITIES.put("High", 700);
        DEFAULT_PRIORITIES.put("Urgent", 800);
        DEFAULT_PRIORITIES.put("Immediate", 1000);
    }

    private String boardId;
    private String boardName;

    public TrelloConfig() {
        super(DEFAULT_PRIORITIES);
    }

    public String getBoardId() {
        return boardId;
    }

    public String getBoardName() {
        return boardName;
    }

    public void setBoardId(String boardId) {
        this.boardId = boardId;
    }

    public void setBoardName(String boardName) {
        this.boardName = boardName;
    }
}
